ls1intum/Orion
# `Feature`: Enable structured grading instructions in Orion

### Motivation and Context
SGIs are a great feature to speed up the assessment process. Prepared feedbacks can be dragged-and-dropped onto comments to apply them, saving the tutors time to type them out. This feature is now also implemented for Orion's inline comments.

### Description
Needs to be tested with https://github.com/ls1intum/Artemis/pull/4104.

SGIs can now dragged onto the feedback text field. Upon insertion, a documentListener parses them and applies them. Also adds the StructuredGradingInstructionLink, which shows applied SGIs and allows to unlink them.

### Steps for Testing


1.  Install the release as described in [the readme](https://github.com/ls1intum/Orion/blob/master/README.md#testing-of-pull-requests)
2.  Open an exercise in assessment mode, the exercise needs to feature sgis and have unassessed submissions
3.  Try out all features of SGIs:
    1. Dragging and Dropping them onto a comment
    3. The tooltip of the link in the comment should show the description
    4. Remove the link to the SGI
    5. The feedback should be displayed in a label above the custome feedback
    6. Add/Removing an SGI gets reverted by cancelling
4. Ensure the changes are correctly saved by Artemis by saving and opening the same submission in Artemis. Comments from Artemis should show correctly in Orion and vice versa.
5. SGIs should behave like in the mentioned Artemis PR, take a look at the change notes there
5. Ensure SGIs in Artemis still work
6. Try to drop non-SGIs into the comment field, it should insert the text as usual
7. Try to drop non-SGIs into the comment field on Artemis, it should behave like before
8. Also think of other use cases of SGIs

#### Code Review
#### Manual Tests
